build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -6,16 +6,67 @@
 
 set(CUDA_HEADERS_API_ROOT "${IREE_ROOT_DIR}/third_party/cuda/include")
 
-external_cc_library(
-  PACKAGE
-    cuda_headers
-  NAME
-    cuda_headers
-  ROOT
-    ${CUDA_HEADERS_API_ROOT}
-  HDRS
-    "cuda.h"
-  INCLUDES
-    ${CUDA_HEADERS_API_ROOT}
-)
+unset(CUDA_INCLUDE_DIR CACHE)
+unset(CUPTI_INCLUDE_DIR CACHE)
```

Comment:
Oh boo, looks like the NO_CACHE option to find_path is only in cmake 3.21 :-/ Why do we need to unset these though? They seem like something that's pretty reasonable to cache (when not debugging this particular configuration), since cuda shouldn't be moving around a lot on someone's machine, and if it is they can reconfigure

Reply:
`find_path` (https://cmake.org/cmake/help/latest/command/find_path.html) has this wonderful behavior where if a variable is already set, it does absolutely nothing transparently. I really cannot fathom why it works this way but it was very confusing time sink when repeated updates to configuration wouldn't correctly set paths until I nuked my cmake cache. 

From what I can tell, if incorrect variables get cached, find_path doesn't work the second time. Perhaps I can just unset without the `NO_CACHE` option but your mileage may vary.

Reply:
Yeah that all makes sense for messing with this. I think now that it's stable though, it makes sense to cache the location of cuda. The reason `find_path` works like that is that it's an expensive operation that is best to avoid repeating. If you want to make things a bit clearer, you could add log lines that trigger if the variable is already defined and note that the previous value is being used.

Reply:
perr @ScottTodd  suggestion, I'm using the `CUDA::cupti` target as the conditional instead of `find_path`.